### PR TITLE
frontend/refactor/ #3209 Refactor flow to reduce API calls

### PR DESCRIPTION
**Pull request by Vignesh-772**
## Type of Change

Refactored flow to reduce Profile API call and Starting Location service.

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [x] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-driver/src/Flow.purs [hunk 1]
```diff
@@ -416,15 +323,72 @@ getDriverInfoFlow = do
...
+      pure $ false
+    Just _ -> do
+      modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen { props {rcActive = true, rcDeactivePopup = false}})
+      pure $ true
+
+checkStatusAndStartLocationUpdates :: FlowBT String Unit
+checkStatusAndStartLocationUpdates = do
+  (GlobalState globalstate) <- getState 
+  let (GetDriverInfoResp getDriverInfoResp) = globalstate.globalProps.driverInformation
+  let isNoDriverMode = isNothing getDriverInfoResp.mode
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Flow.purs:384]: Remove extra lets

File: Frontend/ui-driver/src/Flow.purs [hunk 2]
```diff
@@ -109,6 +109,7 @@ import Types.ModifyScreenState (modifyScreenState, updateStage)
 
 baseAppFlow :: Boolean -> Maybe Event -> FlowBT String Unit
 baseAppFlow baseFlow event = do
+    _ <- pure $ spy "dfbdfv" "DFVdfv"
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Flow.purs:112]: @Vignesh-772 remove spy statements

**Review by Rohit4997 — CHANGES_REQUESTED** [review]

**Review by Rohit4997 — DISMISSED** [review]

**Review by ashishsingh101 — APPROVED** [review]

File: Frontend/ui-driver/src/Screens/ReportIssueChatScreen/Controller.purs [hunk 3]
```diff
@@ -148,8 +148,7 @@ eval BackPressed state =
 eval AfterRender state =
   continue state
 
-eval SubmitIssue state =
-  exit $ UploadIssue state
+eval SubmitIssue state = if state.props.submitIsInProgress then continue state else updateAndExit state{props{submitIsInProgress = true}} $ UploadIssue state{props{submitIsInProgress = true}}
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/ReportIssueChatScreen/Controller.purs:151]: store state{props{submitIsInProgress} " inside newState

File: Frontend/ui-driver/src/Screens/ReportIssueChatScreen/View.purs [hunk 4]
```diff
@@ -310,6 +310,8 @@ submitView push state =
      ]
     ]
   ]
+isEnableSubmitBtn :: ReportIssueChatScreenState -> Boolean
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/ReportIssueChatScreen/View.purs:313]: rename this function.

File: Frontend/ui-driver/src/Screens/HomeScreen/View.purs [hunk 5]
```diff
@@ -1167,9 +1168,13 @@ rideActionModelView push state =
 getPeekHeight :: HomeScreenState -> Int
 getPeekHeight state = 
   let headerLayout = runFn1 JB.getLayoutBounds $ EHC.getNewIDWithTag "rideActionHeaderLayout"
-      labelLayout = runFn1 JB.getLayoutBounds $ EHC.getNewIDWithTag "rideActionLabelLayout"
+      labelLayout =  runFn1 JB.getLayoutBounds $ EHC.getNewIDWithTag "rideActionLabelLayout"
       contentLayout = runFn1 JB.getLayoutBounds $ EHC.getNewIDWithTag "rideActionLayout"
-    in headerLayout.height  + contentLayout.height + (if RideActionModal.isSpecialRide (rideActionModalConfig state) then (labelLayout.height + 6) else 0)
+      pixels = runFn1 HU.getPixels ""
+      density = (runFn1 HU.getDefaultDensity "")/  defaultDensity
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/HomeScreen/View.purs:1174]: rename the function to device default density.

**Review by MercyQueen — APPROVED** [review]

**Review by Rohit4997 — APPROVED** [review]

**Pull request merged**